### ISSUE TITLE
[BUG] `SlidingWindowSplitter` and  `ExpandingWindowSplitter` only returns a single item for test split

### ISSUE THREAD

**Issue by e-alizadeh** [alan-turing-institute/sktime#3071]
**Describe the bug**
When splitting a dataset based on `SlidingWindowSplitter` or `ExpandingWindowSplitter`, the test set for splits has only the last test index instead of an array of indices. The train set is as expected. Check below for a sample output of the issue.

**To Reproduce**

```python
import pandas as pd
from sktime.forecasting.model_selection import SlidingWindowSplitter, ExpandingWindowSplitter

df = pd.DataFrame(list(range(0, 50)))

sliding_splitter = SlidingWindowSplitter(fh=5, window_length=10, step_length=1)
s = list(sliding_splitter.split(df))
print(f"First split of SlidingWindowSplitter: {s[0]}")

expanding_splitter = ExpandingWindowSplitter(fh=5, initial_window=5, step_length=2)
ss = list(expanding_splitter.split(df))
print(f"First split of ExpandingWindowSplitter: {ss[0]}")
```
Above will result in 
```
>>> First split of SlidingWindowSplitter: (array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([14]))
>>> First split of ExpandingWindowSplitter: (array([0, 1, 2, 3, 4]), array([9]))
```

**Expected behavior**
For `SlidingWindowSplitter` the first split based on give arguments must be 
`(array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([10, 11, 12, 13, 14])`

For `ExpandingWindowSplitter` the first split based on give arguments must be 
`(array([0, 1, 2, 3, 4]), array([5, 6, 7, 8, 9]))`

**Additional context**

**Versions**
<details>

System:
    python: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0]
executable: <removed>
   machine: Linux-5.10.60.1-microsoft-standard-WSL2-x86_64-with-glibc2.17

Python dependencies:
          pip: 22.1.2
   setuptools: 61.2.0
      sklearn: 1.1.1
       sktime: 0.12.1
  statsmodels: 0.13.2
        numpy: 1.22.4
        scipy: 1.8.1
       pandas: 1.4.3
   matplotlib: 3.5.2
       joblib: 1.1.0
        numba: 0.55.2
     pmdarima: 1.8.5
      tsfresh: 0.19.0
</details>

**Comment by khrapovs** [comment]
@e-alizadeh Thanks a lot for reporting this! Actually, I wouldn't say this is a bug, but rather a badly documented feature. Test window will include only those observations that are explicitly listed in the forecasting horizon. The desired result is achieved by providing `np.arange(1, 6)` to the `fh` argument:
```
import numpy as np
import pandas as pd
from sktime.forecasting.model_selection import SlidingWindowSplitter, ExpandingWindowSplitter

df = pd.DataFrame(list(range(0, 50)))

sliding_splitter = SlidingWindowSplitter(fh=np.arange(1, 6), window_length=10, step_length=1)
s = list(sliding_splitter.split(df))
print(f"First split of SlidingWindowSplitter: {s[0]}")

expanding_splitter = ExpandingWindowSplitter(fh=np.arange(1, 6), initial_window=5, step_length=2)
ss = list(expanding_splitter.split(df))
print(f"First split of ExpandingWindowSplitter: {ss[0]}")
```
I will try to add a few lines to the docstring of the splitters in the coming days. I hope this would be sufficient. Changing splitters behaviour to expand test window up to the maximum forecasting horizon is less explicit and therefore not a very good idea. What do you think?

**Comment by fkiraly** [comment]
Possible alternative approach:
adding an argument `forecast_window_length`, where `forecast_window_length=x` is equivalent to passing `fh=np.arange(1, x+1)`, and not both arguments `fh` and `forecast_window_length` can be passed (exactly one must be given)

**Comment by e-alizadeh** [comment]
Thanks for looking into this. The main reason why I thought this is a bug is that for both splitters, my examples above are similar to the ones in the documentation as can be seen in the blow screenshot.
![Screen Shot 2022-07-22 at 9 07 06 AM](https://user-images.githubusercontent.com/30320825/180445536-fd0dc8d5-e742-4594-9dfb-fe971d2b2dd8.png)

On a different note, having actual implementation examples in the docstring would help a lot. The User Guides and Tutorials do not have examples of all transformers, and having a sample implementation for each object in the same page would help a lot!

**Comment by fkiraly** [comment]
ah, I see!
Was it Donald Knuth who said, if the docstring and the code do not agree, there is a good chance that both are wrong?
(probably paraphrasing this badly)

**Comment by khrapovs** [comment]
Offtopic of course, but that's one of the reasons I ignore docstrings whenever possible. They rot immediately after being written.

**Comment by fkiraly** [comment]
>Offtopic of course, but that's one of the reasons I ignore docstrings whenever possible. They rot immediately after being written.

😱 🤮 😱 why do you hurt me so, @khrapovs

**Comment by khrapovs** [comment]
@e-alizadeh Please, check the changes in https://github.com/alan-turing-institute/sktime/pull/3075. I will edit the text as you like.

**Comment by e-alizadeh** [comment]
Thanks @khrapovs. I've checked the changes and they look good to me.
